tests/test_rules.py

```diff
@@ -1,269 +1,357 @@
 # -*- coding: utf-8 -*-
+import json
+import logging
 import datetime as dt
+from collections import defaultdict
 
-from doorman.models import Rule
 from doorman.rules import (
-    BaseRule,
-    BlacklistRule,
-    RuleMatch,
-    WhitelistRule,
+    BaseCondition,
+    EqualCondition,
+    Network,
+    RuleInput,
 )
 
 
-class TestBaseRule:
+DUMMY_INPUT = RuleInput(result_log={}, node={})
 
-    def test_will_filter_node_name(self, FakeBaseRule):
-        rule = FakeBaseRule(config={'node_name': 'yes'})
 
-        # This should not succeed due to a host_identifier that does not match
-        rule.handle_log_entry({}, {'host_identifier': 'no'})
-        assert len(rule.calls) == 0
+class TestNetwork:
 
-        # This should succeed since the host_identifier does match.  Note that
-        # we explicitly don't care about the hostIdentifier value in the query
-        # results.
-        now = dt.datetime.utcnow()
-        rule.handle_log_entry({
-            'data': [
-                {
-                    "diffResults": {
-                        "added": [{'op': 'added'}],
-                        "removed": "",
-                    },
-                    "name": "fake",
-                    "hostIdentifier": "hostname.local",
-                    "calendarTime": "%s %s" % (now.ctime(), "UTC"),
-                    "unixTime": now.strftime('%s')
-                }
-            ]
-        }, {'host_identifier': 'yes'})
-        assert len(rule.calls) == 1
+    def test_will_cache_condition_instances(self):
+        class TestCondition(BaseCondition):
+            pass
 
+        network = Network()
+        one = network.make_condition(TestCondition)
+        two = network.make_condition(TestCondition)
 
-class TestEachResultRule:
-    def setup_method(self, _method):
-        self.now = dt.datetime.utcnow()
-        self.fake_data = {
-            'data': [
-                {
-                    "diffResults": {
-                        "added": [{'op': 'added'}],
-                        "removed": [{'op': 'removed'}],
-                    },
-                    "name": "fake",
-                    "hostIdentifier": "hostname.local",
-                    "calendarTime": "%s %s" % (self.now.ctime(), "UTC"),
-                    "unixTime": self.now.strftime('%s')
-                },
-            ],
+        assert one is two
+
+    def test_will_parse_basic(self):
+        query = json.loads("""
+        {
+          "condition": "AND",
+          "rules": [
+            {
+              "id": "column",
+              "field": "column",
+              "type": "string",
+              "input": "text",
+              "operator": "column_equal",
+              "value": [
+                "model_id",
+                "5500"
```

Comment:
How is this possible? An AND condition with a column_equal to a value of `["model_id", "5500"]`?